homeassistant/components/modbus/modbus.py

```diff
@@ -111,7 +111,8 @@ async def async_modbus_setup(
 ):
     """Set up Modbus component."""
 
-    hass.data[DOMAIN] = hub_collect = {}
+    hub_collect = {}
+    hass.set_domain(DOMAIN, hub_collect)
```

Comment:
It's not just this though. Some integrations have multiple keys in there, especially in combination with things like the `@singleton` helper. 

I actually don't see too often talk of people misunderstanding hass.data.

Reply:
The parameter is type Any so anything goes.

However this was just a proposal, and if you do not see it as a problem I will simply close it.